src/sys/eventfd.rs

```diff
@@ -10,8 +10,58 @@ libc_bitflags! {
     }
 }
 
+#[deprecated(since = "0.27.0", note = "Use EventFd::value_and_flags() instead")]
 pub fn eventfd(initval: libc::c_uint, flags: EfdFlags) -> Result<OwnedFd> {
     let res = unsafe { libc::eventfd(initval, flags.bits()) };
 
     Errno::result(res).map(|r| unsafe { OwnedFd::from_raw_fd(r) })
 }
+
+#[derive(Debug)]
+pub struct EventFd(pub OwnedFd);
+impl EventFd {
+    /// [`EventFd::value_and_flags`] with `init_val = 0` and `flags = EfdFlags::empty()`.
+    pub fn new() -> Result<Self> {
+        let res = unsafe { libc::eventfd(0, EfdFlags::empty().bits()) };
+        Errno::result(res).map(|r| Self(unsafe { OwnedFd::from_raw_fd(r) }))
+    }
+    /// Constructs [`EventFd`] with the given `init_val` and `flags`.
+    /// 
+    /// Wrapper around [`libc::eventfd`].
+    pub fn value_and_flags(init_val: u32, flags: EfdFlags) -> Result<Self> {
+        let res = unsafe { libc::eventfd(init_val, flags.bits()) };
+        Errno::result(res).map(|r| Self(unsafe { OwnedFd::from_raw_fd(r) }))
+    }
+    /// [`EventFd::value_and_flags`] with `init_val = 0` and given `flags`.
+    pub fn flags(flags: EfdFlags) -> Result<Self> {
+        let res = unsafe { libc::eventfd(0, flags.bits()) };
+        Errno::result(res).map(|r| Self(unsafe { OwnedFd::from_raw_fd(r) }))
+    }
+    /// [`EventFd::value_and_flags`] with given `init_val` and `flags = EfdFlags::empty()`.
+    pub fn value(init_val: u32) -> Result<Self> {
+        let res = unsafe { libc::eventfd(init_val, EfdFlags::empty().bits()) };
+        Errno::result(res).map(|r| Self(unsafe { OwnedFd::from_raw_fd(r) }))
+    }
+    /// [`EventFd::write`] with `1`.
+    pub fn arm(&self) -> Result<usize> {
+        unistd::write(self.0.as_raw_fd(),&1u64.to_ne_bytes())
+    }
+    /// [`EventFd::write`] with `0`.
+    pub fn defuse(&self) -> Result<usize> {
+        unistd::write(self.0.as_raw_fd(),&0u64.to_ne_bytes())
+    }
+    /// Writes a given `value` to the file descriptor.
+    pub fn write(&self, value: u64) -> Result<usize> {
```

Comment:
What is the meaning of writing a value to the file descriptor?

Reply:
This is literally what it is doing, although I would agree its not the clearest documentation although I'm not sure what to replace it with.

https://man7.org/linux/man-pages/man2/write.2.html

> write() writes up to count bytes from the buffer starting at buf to the file referred to by the file descriptor fd.

Reply:
But what is even the purpose of writing to an eventFD?  Is there any reason why anybody would ever want to do that?

Reply:
When you poll on an `EventFd` you can consider that the event fires when a non-zero value can be read from the `EventFd`. By writing a value to the `EventFd` you are setting it up to trigger `value` events when polled.

If you write `2` here then poll 3 times on the `EventFd` the 1st 2 polls will return and the 3rd will wait.

Reply:
Ok.  Could you please rewrite the docs then, in terms of "arming" instead of "writing".  We should strive to be as high-level as possible while remaining zero-cost.  Also, is there any point in passing through the return value of `unistd::write`?

Reply:
I've updated the docs.